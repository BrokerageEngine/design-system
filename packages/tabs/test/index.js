import React from 'react'
import { mount, shallow } from 'enzyme'
import { Box, Link, Text, ThemeProvider } from 'pcln-design-system'
import renderer from 'react-test-renderer'
import Tabs from '../src'

const withTheme = Component => <ThemeProvider>{Component}</ThemeProvider>

const props = {
  activeColor: 'darkBlue',
  defaultColor: 'blue',
  tabs: [
    {
      title: 'Tab 1'
    },
    {
      title: 'Tab 2'
    },
    {
      title: 'Tab 3'
    }
  ]
}

const childElements = [
  <Box>Child 1</Box>,
  <Box>Child 2</Box>,
  <Box>Child 3</Box>
]

const mappedProps = [
  {
    heading: 'Mapped content 1',
    body: 'Body Text 1'
  },
  {
    heading: 'Mapped content 2',
    body: 'Body Text 2'
  },
  {
    heading: 'Mapped content 3',
    body: 'Body Text 3'
  }
]

const ExampleTab = props => (
  <Flex flexDirection="column">
    <Heading>{props.heading}</Heading>
    <Box>{props.body}</Box>
  </Flex>
)

const mockHotelProps = {
  activeColor: 'darkBlue',
  defaultColor: 'blue',
  tabs: [
    {
      title: 'Tab 1',
      content: {
        confirmationNumber: '123-TEST'
      }
    },
    {
      title: 'Tab 2',
      content: {
        hotelName: 'Venetian Las Vegas'
      }
    },
    {
      title: 'Tab 3',
      content: {
        importantInfo: 'Free wifi'
      }
    }
  ]
}

const ManageTab = props => <Box>{props.confirmationNumber}</Box>

const RoomTab = props => <Box>{props.hotelName}</Box>

const ImportantInfoTab = props => <Box>{props.importantInfo}</Box>

// content is not tied to tabs array

const HardCodedTabContent = withTheme(
  <Tabs {...props}>
    <Box>Test 1</Box>
    <Box>Test 2</Box>
    <Box>Test 3</Box>
  </Tabs>
)

const MappedElementsTabs = withTheme(
  <Tabs {...props}>{childElements.map(child => child)}</Tabs>
)

const MappedPropsTabs = withTheme(
  <Tabs {...props}>
    {mappedProps.map((content, i) => (
      <ExampleTab key={i} {...content} />
    ))}
  </Tabs>
)

const HotelTabs = withTheme(
  <Tabs {...mockHotelProps}>
    <ManageTab {...mockHotelProps.tabs[0].content} />
    <RoomTab {...mockHotelProps.tabs[1].content} />
    <ImportantInfoTab {...mockHotelProps.tabs[2].content} />
  </Tabs>
)

describe('Tab Navigation', () => {
  test('renders with hardcoded elements', () => {
    const wrapper = mount(HardCodedTabContent)
    const tree = renderer.create(wrapper).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders with mapped elements', () => {
    const wrapper = mount(MappedElementsTabs)
    const tree = renderer.create(wrapper).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders with mapped props', () => {
    const wrapper = mount(MappedPropsTabs)
    const tree = renderer.create(wrapper).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders with composite components consuming props', () => {
    const wrapper = mount(HotelTabs)
    const tree = renderer.create(wrapper).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('default activeTab should be first index of tabs array', () => {
    const wrapper = mount(HotelTabs)
    const manageTabHeading = wrapper.find('Styled(Text)').first()
    // const manageTabContent = wrapper.find('ManageTab')
    console.log(wrapper.debug())
    expect(manageTabHeading.prop('isActive')).toBe(true)
    // expect(manageTabContent).toHaveLength(1)
  })

  test('on click of 2nd tab title should switch activeTab to index 1 of tab array', () => {
    const wrapper = mount(ThemedTabs)
    const roomsTabHeading = wrapper.find('Styled(Text)').at(1)
    expect(roomsTabHeading.prop('isActive')).toBe(false)

    // write test to make sure 2nd child component is displayed
    // write test to test color change of active Tab

    roomsTabHeading.simulate('click')
    const manageTabHeading = wrapper.find('Styled(Text)').first()
    const manageTabContent = wrapper.find('ManageTab')
    const postClickRoomsTab = wrapper.find('Styled(Text)').at(1)
    const hotelImage = wrapper.find('Image')
    expect(manageTabHeading.prop('isActive')).toBe(false)
    expect(manageTabContent).toHaveLength(0)
    expect(postClickRoomsTab.prop('isActive')).toBe(true)
    expect(hotelImage.prop('src')).toBe(
      'https://mobileimg.priceline.com/htlimg/50/50564/thumbnail-150-square.jpg'
    )
  })

  test('on click of 2nd tab title should display 2nd child of the component', () => {})

  test("on click of tab title should update active tab title's styles", () => {})

  test('tab title text defaultColor and activeColor can be customized via props', () => {})
})
