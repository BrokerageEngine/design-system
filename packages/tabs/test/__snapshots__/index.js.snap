// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Navigation renders with composite components consuming props 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 8px;
  padding-left: 4px;
  padding-right: 4px;
  background-color: #f6f8fa;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
  padding-bottom: 16px;
  background-color: #fff;
}

.c3 {
  margin-left: 16px;
  margin-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #049;
}

.c5 {
  margin-left: 16px;
  margin-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #007aff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c0 * {
  box-sizing: border-box;
}

.c2 {
  border-bottom: 4px solid darkBlue;
  cursor: pointer;
}

.c4 {
  border-bottom: none;
  cursor: pointer;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2 c3"
      color="darkBlue"
      onClick={[Function]}
    >
      Tab 1
    </div>
    <div
      className="c4 c5"
      color="blue"
      onClick={[Function]}
    >
      Tab 2
    </div>
    <div
      className="c4 c5"
      color="blue"
      onClick={[Function]}
    >
      Tab 3
    </div>
  </div>
  <div
    className="c6"
  >
    <div
      className=""
    >
      123-TEST
    </div>
  </div>
</div>
`;

exports[`Tab Navigation renders with hardcoded elements 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 8px;
  padding-left: 4px;
  padding-right: 4px;
  background-color: #f6f8fa;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
  padding-bottom: 16px;
  background-color: #fff;
}

.c3 {
  margin-left: 16px;
  margin-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #049;
}

.c5 {
  margin-left: 16px;
  margin-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #007aff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c0 * {
  box-sizing: border-box;
}

.c2 {
  border-bottom: 4px solid darkBlue;
  cursor: pointer;
}

.c4 {
  border-bottom: none;
  cursor: pointer;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2 c3"
      color="darkBlue"
      onClick={[Function]}
    >
      Tab 1
    </div>
    <div
      className="c4 c5"
      color="blue"
      onClick={[Function]}
    >
      Tab 2
    </div>
    <div
      className="c4 c5"
      color="blue"
      onClick={[Function]}
    >
      Tab 3
    </div>
  </div>
  <div
    className="c6"
  >
    <div
      className=""
    >
      Test 1
    </div>
  </div>
</div>
`;

exports[`Tab Navigation renders with mapped elements 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 8px;
  padding-left: 4px;
  padding-right: 4px;
  background-color: #f6f8fa;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
  padding-bottom: 16px;
  background-color: #fff;
}

.c3 {
  margin-left: 16px;
  margin-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #049;
}

.c5 {
  margin-left: 16px;
  margin-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #007aff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c0 * {
  box-sizing: border-box;
}

.c2 {
  border-bottom: 4px solid darkBlue;
  cursor: pointer;
}

.c4 {
  border-bottom: none;
  cursor: pointer;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2 c3"
      color="darkBlue"
      onClick={[Function]}
    >
      Tab 1
    </div>
    <div
      className="c4 c5"
      color="blue"
      onClick={[Function]}
    >
      Tab 2
    </div>
    <div
      className="c4 c5"
      color="blue"
      onClick={[Function]}
    >
      Tab 3
    </div>
  </div>
  <div
    className="c6"
  >
    <div
      className=""
    >
      Child 1
    </div>
  </div>
</div>
`;

exports[`Tab Navigation renders with mapped props 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 8px;
  padding-left: 4px;
  padding-right: 4px;
  background-color: #f6f8fa;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
  padding-bottom: 16px;
  background-color: #fff;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  margin-left: 16px;
  margin-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #049;
}

.c5 {
  margin-left: 16px;
  margin-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #007aff;
}

.c8 {
  font-size: 24px;
  margin: 0px;
  font-weight: 400;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c0 * {
  box-sizing: border-box;
}

.c2 {
  border-bottom: 4px solid darkBlue;
  cursor: pointer;
}

.c4 {
  border-bottom: none;
  cursor: pointer;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2 c3"
      color="darkBlue"
      onClick={[Function]}
    >
      Tab 1
    </div>
    <div
      className="c4 c5"
      color="blue"
      onClick={[Function]}
    >
      Tab 2
    </div>
    <div
      className="c4 c5"
      color="blue"
      onClick={[Function]}
    >
      Tab 3
    </div>
  </div>
  <div
    className="c6"
  >
    <div
      className="c7"
    >
      <h3
        className="c8"
        fontSize={4}
      >
        Mapped content 1
      </h3>
      <div
        className=""
      >
        Body Text 1
      </div>
    </div>
  </div>
</div>
`;
